src/device.rs

```diff
@@ -93,6 +101,72 @@ pub unsafe extern "C" fn wgpuAdapterGetProperties(
     }
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn wgpuAdapterGetLimits(
+    adapter: id::AdapterId,
+    limits: &mut native::WGPUSupportedLimits,
+) -> bool {
+    let result = gfx_select!(adapter => GLOBAL.adapter_limits(adapter));
+    match result {
+        Ok(wgt_limits) => write_limits_struct(wgt_limits, limits),
+        _ => panic!("Calling wgpuAdapterGetLimits() on an invalid adapter."),
+    }
+    return false; // todo: what is the purpose of this return value?
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn wgpuDeviceGetLimits(
+    device: id::DeviceId,
+    limits: &mut native::WGPUSupportedLimits,
+) -> bool {
+    let result = gfx_select!(device => GLOBAL.device_limits(device));
+    match result {
+        Ok(wgt_limits) => write_limits_struct(wgt_limits, limits),
+        _ => panic!("Calling wgpuDeviceGetLimits() on an invalid device."),
+    }
+    return false;
+}
+
+fn write_limits_struct(
+    wgt_limits: wgt::Limits,
+    supported_limits: &mut native::WGPUSupportedLimits,
+) {
+    let mut limits = supported_limits.limits; // This makes a copy - we copy back at the end
```

Comment:
There is probably a way to write this such that `limits` is a reference instead of a copy, so that the final line of this function is not needed.